### PR TITLE
SMC: replace multinomial sampling with systematic sampling

### PR DESCRIPTION
**Pull request by aloctavodia**
Empirically, systematic sampling has been shown to be systematically ;-) better than multinomial sampling when used for SMC. 

**Checklist**
+ [x] Explain important implementation details 👆
+ [x] Make sure that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html).
+ [x] Link relevant issues (preferably in [nice commit messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html))
+ [x] Are the changes covered by tests and docstrings?
+ [x] Fill out the short summary sections 👇

## Major / Breaking Changes
- ...

## Bugfixes / New features
-  SMC: replace multinomial sampling with systematic sampling

## Docs / Maintenance
- ...

### CODE REVIEW

File: pymc/smc/smc.py [hunk 1]
```diff
@@ -546,6 +544,36 @@ def sample_settings(self):
...
+        The weights should be probabilities and the total sum should be 1.
+
+    Returns
+    -------
+    new_indices: array
+        A vector of indices in the interval 0, ..., len(normalized_weights)
+    """
+    lnw = len(weights)
+    arange = np.arange(lnw)
+    uniform = (np.random.rand(1) + arange) / lnw
```
Comment (ricardoV94) [inline pymc/smc/smc.py]: This should use a numpy generator for seeding, which is passed to the function (`self.rng` in the case of SMC kernels)

File: pymc/smc/smc.py [hunk 2]
```diff
@@ -546,6 +544,36 @@ def sample_settings(self):
         return stats
 
 
+def systematic(weights):
```
Comment (ricardoV94) [inline pymc/smc/smc.py]: Can we give this a more expressive name? And also test it directly?

**Review by ricardoV94 — CHANGES_REQUESTED** [review]

**Review by ricardoV94 — APPROVED** [review]

**Comment by ricardoV94** [comment]
@aloctavodia you forgot to squash the two commits when merging. No biggie :)

**Comment by aloctavodia** [comment]
Thanks for the remainder. I completely forgot about that.

**Pull request merged**